test/fake_slack.ml

```diff
@@ -0,0 +1,98 @@
+(* A (currently pretty useless) fake slack implementation to run tests
+   against. *)
+open Lwt
+open Cohttp
+open Cohttp_lwt_unix
+
+
+let channels_json = Yojson.Safe.from_file "channels.json"
+let new_channel_json = Yojson.Safe.from_file "new_channel.json"
+let authed_json = Yojson.Safe.from_file "authed.json"
+
+let json_fields = function
+  | `Assoc fields -> fields
+  | _ -> failwith "Can't parse test json."
+
+
+let resp ok fields =
+  let body = `Assoc (("ok", `Bool ok) :: fields) |> Yojson.Safe.to_string in
+  Server.respond_string ~status:`OK ~body ()
+
+let ok_resp fields = resp true fields
+
+let err_resp err fields = resp false (("error", `String err) :: fields)
+
+let get_arg_opt arg req =
+  Uri.get_query_param (Request.uri req) arg
+
+let get_arg_default arg default req =
+  match get_arg_opt arg req with
+  | Some x -> x
+  | None -> default
+
+let get_arg arg req =
+  match get_arg_opt arg req with
+  | Some x -> x
+  | None -> failwith @@ "Mandatory arg " ^ arg ^ " not given."
+
+let check_auth f req body =
+  match get_arg "token" req with
+  | "xoxp-testtoken" -> f req body
```

Comment:
`fake_slack` is currently very specific to the test data it's working with. It probably needs to become more general, but that ties into the difference between unit/functional/integration tests and how we handle the much less predictable data we get when talking to the real slack API.